src/solver/utils/filename.h

```diff
@@ -1,10 +1,68 @@
 #pragma once
 
 #include <string>
-#include <yuni/core/string.h>
+#include <memory>
 
-std::string getFilenameWithExtension(const YString& prefix,
-                                     const YString& extension,
-                                     uint year,
-                                     uint week,
-                                     uint optNumber = 0);
+
+/*
+    MPS and criterion generated files wear the same time interval sequence in their names.
+    This time interval sequence is either <year>-<week> (for a weekly optimization),
+    or <year>-<week>-<day> (for a daily optimization). 
+    So, depending on whether the optimization is weekly or daily, the files are named either :
+    - problem-<year>-<week>--optim-nb-<n>.mps
+    - problem-<year>-<week>-<day>--optim-nb-<n>.mps
+    
+    The following class hierarchy intend to reprensent these time interval sequences (weekly or daily),
+    and it turns them into a string when building the names of the previous files.
+    This hierarchy is designed for polymorphism : whetever the time interval an object of
+    that hierarchy represents, it is passed as a base class argument, and the right 'to_string()'
+    function is called.
+*/
+// ------------------------------------
+// Optimization period : base class
+// ------------------------------------
+class optPeriodAsString
+{
+public:
+    virtual std::string to_string() const = 0;
+    virtual ~optPeriodAsString() = default;
+};
```

Comment:
Move this class declaration to another header file.

Reply:
Done